validate.go

```diff
@@ -54,9 +55,13 @@ func ValidateCid(c cid.Cid) error {
 		return ErrPossiblyInsecureHashFunction
 	}
 
-	if pref.MhType != mh.ID && pref.MhLength < minimumHashLength {
+	if pref.MhType != mh.IDENTITY && pref.MhLength < minimumHashLength {
 		return ErrBelowMinimumHashLength
 	}
 
+	if pref.MhType == mh.BLAKE3 && pref.MhLength != 32 {
+		return ErrUnsupportedBLAKE3Length
+	}
```

Comment:
@Jorropo WDYT about doing something like this instead:
```suggestion
	if pref.MhType != mh.IDENTITY && pref.MhLength > maximumHashLength {
		return ErrAboveMaximumHashLength
	}
```

where `maximumHashLength` is 128 bytes which is AFACIT the biggest we have registered in the table https://github.com/multiformats/multicodec/blob/master/table.csv#L476.

cc @Stebalien @Kubuxu if you think that limit should be larger or smaller (e.g. only 64 bytes and matching the largest currently in go-verifcid)?

Reply:
Sgtm but that wont resolve `go-multihash` issue no ?

Reply:
Which issue, I might be confused.

Reply:
If i give you a 128 bytes big blake3 CID you need to create a 128 bytes big blake3 digest to compare against. Which we dont do afaik.

Reply:
hey @aschmahmann see comment above for why i'm like 90% sure this pattern can't work: https://github.com/ipfs/go-verifcid/pull/15#discussion_r915263122

Reply:
there is also no "check hash" functionality in go-multihash to correct in this way- there's only "make hash" "make another hash" "compare the two bytestrings"

Reply:
@laudiacay look at this code:

https://github.com/ipfs/go-cid/blob/b2064d74a8b098193b316689a715cdf4e4934805/cid.go#L532

It's `go-cid` passing the expected length to `go-multihash`, right now the only logic in multihash's side is to truncate.
https://github.com/multiformats/go-multihash/blob/89bf86c75b6b66226f63e22db775a72172970360/sum.go#L70 (truncate)

But IMO it would be easy to add a length argument to the hasher.
Or maybe make a second map of hashers for variable sized hashers.

It might break a few things up top but IMO nothing we can't fix easily.

Reply:
@Jorropo https://github.com/multiformats/go-multihash/pull/159

Reply:
But please see my comments on the other thread.